sklearnex/ensemble/_forest.py

```diff
@@ -1122,6 +1128,7 @@ def predict(self, X):
     predict.__doc__ = sklearn_ForestRegressor.predict.__doc__
 
 
+@support_n_jobs
```

Comment:
Please remove @support_n_jobs for all of the classes in _forest.py, n_jobs has been supported by an inherited sklearn class since at least 0.24 https://github.com/scikit-learn/scikit-learn/blob/0.24.X/sklearn/ensemble/_forest.py#L204. All the actions in that function are unnecessary. If it still must be included, use ForestClassifier and ForestRegressor classes instead.

Reply:
It wasn't clear this decorator is targeted to control n_jobs behaviour in all estimators patched by oneDAL so I renamed it to `control_n_jobs` and extended comment.

Reply:
I now understand the necessity of doing so. For the patched estimators in _forest, they are thin APIs over ForestClassifier and ForestRegressor. I believe for maintainability purposes, it is best to switch these decorators to those classes. I will agree to the changes to the direct patched algos if that was the agreed upon rule in the arch meeting. In general, this will then need to be written somewhere for future introduced algos.